### PR TITLE
refactored basic auth

### PR DESCRIPTION
**Pull request by sinamna**
## Summary by CodeRabbit

- **Refactor**
  - Streamlined the authentication process by integrating new configuration fields.
  - Enhanced the initialization of authentication variables for improved clarity and maintainability.

- **Chores**
  - Removed unused code and dependencies to simplify the codebase and reduce potential errors.

- **Bug Fixes**
  - Adjusted the authentication workflow to prevent potential issues with credential and configuration management.

### CODE REVIEW

File: internal/controller/basic_authenticator/basicauthenticator_controller.go [hunk 1]
```diff
@@ -46,9 +49,15 @@ func (r *BasicAuthenticatorReconciler) Reconcile(ctx context.Context, req ctrl.R
...
 	logger.Info("reconcile triggered")
 	logger.Info(req.String())
+	r.setupVars(req)
 	return r.Provision(ctx, req)
 }
 
+func (r *BasicAuthenticatorReconciler) setupVars(request ctrl.Request) {
+	r.basicAuthenticatorNamespace = request.Namespace
+	// add others
+}
```
[the remark this answers is not in the document]
Comment (hoptical) [reply internal/controller/basic_authenticator/basicauthenticator_controller.go:59]: @coderabbitai It's not needed.
@sinamna Please use `initVars` instead to make it look like our other repositories.

File: internal/controller/basic_authenticator/basicauthenticator_controller.go [hunk 2]
```diff
@@ -46,9 +49,15 @@ func (r *BasicAuthenticatorReconciler) Reconcile(ctx context.Context, req ctrl.R
 	logger := log.FromContext(ctx)
 	logger.Info("reconcile triggered")
 	logger.Info(req.String())
+	r.setupVars(req)
 	return r.Provision(ctx, req)
```
[the remark this answers is not in the document]
Comment (hoptical) [reply internal/controller/basic_authenticator/basicauthenticator_controller.go:53]: @coderabbitai It's not needed

File: internal/controller/basic_authenticator/provision.go [hunk 3]
```diff
@@ -53,9 +53,9 @@ func (r *BasicAuthenticatorReconciler) ensureSecret(ctx context.Context, req ctr
 	if r, err := r.GetLatestBasicAuthenticator(ctx, req, basicAuthenticator); subreconciler.ShouldHaltOrRequeue(r, err) {
 		return r, err
 	}
-	credentialName := basicAuthenticator.Spec.CredentialsSecretRef
+	r.credentialName = basicAuthenticator.Spec.CredentialsSecretRef
```
[the remark this answers is not in the document]
Comment (hoptical) [reply internal/controller/basic_authenticator/provision.go:56]: Agree

File: internal/controller/basic_authenticator/provision.go [hunk 4]
```diff
@@ -67,12 +67,10 @@ func (r *BasicAuthenticatorReconciler) ensureSecret(ctx context.Context, req ctr
...
 
-			credentialName = newSecret.Name
+			r.credentialName = newSecret.Name
 			credentialSecret = *newSecret
-			basicAuthenticator.Spec.CredentialsSecretRef = credentialName
-
+			basicAuthenticator.Spec.CredentialsSecretRef = r.credentialName
 			//saving secretName inorder to be used in next steps
-			assignAnnotation(basicAuthenticator, SecretAnnotation, credentialName)
 			err = r.Update(ctx, basicAuthenticator)
```
[the remark this answers is not in the document]
Comment (hoptical) [reply internal/controller/basic_authenticator/provision.go:74]: Not needed

File: internal/controller/basic_authenticator/provision.go [hunk 5]
```diff
@@ -122,17 +113,13 @@ func (r *BasicAuthenticatorReconciler) ensureConfigmap(ctx context.Context, req
...
 		//saving secretName inorder to be used in next steps
-		assignAnnotation(basicAuthenticator, ConfigmapAnnotation, authenticatorConfig.Name)
+		r.configMapName = authenticatorConfig.Name
 
-		err = r.Update(ctx, basicAuthenticator)
-		if err != nil {
-			logger.Error(err, "failed to updated basic authenticator")
-			return subreconciler.RequeueWithError(err)
-		}
 		return subreconciler.Requeue()
```
[the remark this answers is not in the document]
Comment (hoptical) [reply internal/controller/basic_authenticator/provision.go:118]: @sinamna Any reason for the requeue?

File: internal/controller/basic_authenticator/provision.go [hunk 6]
```diff
@@ -143,12 +130,7 @@ func (r *BasicAuthenticatorReconciler) ensureConfigmap(ctx context.Context, req
 				return subreconciler.RequeueWithError(err)
 			}
 		}
-		assignAnnotation(basicAuthenticator, ConfigmapAnnotation, authenticatorConfig.Name)
-		err = r.Update(ctx, basicAuthenticator)
-		if err != nil {
-			logger.Error(err, "failed to updated basic authenticator")
-			return subreconciler.RequeueWithError(err)
-		}
+		r.configMapName = authenticatorConfig.Name
```
[the remark this answers is not in the document]
Comment (hoptical) [reply internal/controller/basic_authenticator/provision.go:133]: Not Needed

**Review by hoptical — COMMENTED** [review]
`assignAnnotation` in `utility.go` is not needed anymore.

File: internal/controller/basic_authenticator/provision.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (sinamna) [reply internal/controller/basic_authenticator/provision.go:118]: kinda made a mistake in here. I'll fix it

File: internal/controller/basic_authenticator/provision.go [hunk 7]
```diff
@@ -161,32 +143,29 @@ func (r *BasicAuthenticatorReconciler) ensureDeployment(ctx context.Context, req
...
+	//authenticatorConfigName, configmapExists := basicAuthenticator.ObjectMeta.Annotations[ConfigmapAnnotation]
+	if r.configMapName == "" {
 		return subreconciler.RequeueWithError(defaultError.New("configmap annotation not set"))
 	}
 
-	secretName, secretExists := basicAuthenticator.ObjectMeta.Annotations[SecretAnnotation]
-	if !secretExists {
+	//secretName, secretExists := basicAuthenticator.ObjectMeta.Annotations[SecretAnnotation]
+	if r.credentialName == "" {
 		return subreconciler.RequeueWithError(defaultError.New("secret annotation not set"))
```
[the remark this answers is not in the document]
Comment (sinamna) [reply internal/controller/basic_authenticator/provision.go]: changed error messages

**Comment by sinamna** [comment]
> `assignAnnotation` in `utility.go` is not needed anymore.

done

File: internal/controller/basic_authenticator/provision.go [hunk 8]
```diff
@@ -231,9 +210,9 @@ func (r *BasicAuthenticatorReconciler) CreateDeploymentAuthenticator(ctx context
 	return subreconciler.ContinueReconciling()
```
[the remark this answers is not in the document]
Comment (sinamna) [reply internal/controller/basic_authenticator/provision.go]: done

File: internal/controller/basic_authenticator/provision.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (sinamna) [reply internal/controller/basic_authenticator/provision.go]: done

File: internal/controller/basic_authenticator/provision.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (sinamna) [reply internal/controller/basic_authenticator/provision.go:55]: done

File: internal/controller/basic_authenticator/basicauthenticator_controller.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (sinamna) [reply internal/controller/basic_authenticator/basicauthenticator_controller.go]: done

**Review by hoptical — APPROVED** [review]

**Pull request merged**